File: schemaorggo/businessaudience.go
package schemaorggo

import "encoding/json"

// BusinessAudience see : https://schema.org/BusinessAudience
type BusinessAudience struct {
	Audience

	typeContext

	// NumberOfEmployees see : https://schema.org/numberOfEmployees
	// The number of employees in an organization e.g. business.
	NumberOfEmployees *QuantitativeValue `json:"numberOfEmployees,omitempty"` // types : QuantitativeValue

	// YearlyRevenue see : https://schema.org/yearlyRevenue
	// The size of the business in annual revenue.
	YearlyRevenue *QuantitativeValue `json:"yearlyRevenue,omitempty"` // types : QuantitativeValue

	// YearsInOperation see : https://schema.org/yearsInOperation
	// The age of the business.
	YearsInOperation *QuantitativeValue `json:"yearsInOperation,omitempty"` // types : QuantitativeValue

}

func (v BusinessAudience) MarshalJSONWithTypeContext() ([]byte, error) {
	v.C = "http://schema.org"
	v.T = "BusinessAudience"

	return json.Marshal(v)
}

func (v *BusinessAudience) MarshalJSON() ([]byte, error) {
	if v == nil {
		return []byte("null"), nil
	}

	v.C = "http://schema.org"
	v.T = "BusinessAudience"

	return json.Marshal(*v)
}

File: schemaorgphp/classes/business_audience.php
<?php

class BusinessAudience extends Audience implements JsonSerializable {

	public static $context = 'http://schema.org';
	public static $type = 'BusinessAudience';
	
	/**
	 * The number of employees in an organization e.g. business.
	 * see : https://schema.org/numberOfEmployees
	 * @var \QuantitativeValue|\QuantitativeValue[]
	 */
	public var $number_of_employees;
	
	/**
	 * The size of the business in annual revenue.
	 * see : https://schema.org/yearlyRevenue
	 * @var \QuantitativeValue|\QuantitativeValue[]
	 */
	public var $yearly_revenue;
	
	/**
	 * The age of the business.
	 * see : https://schema.org/yearsInOperation
	 * @var \QuantitativeValue|\QuantitativeValue[]
	 */
	public var $years_in_operation;
	
	public function jsonSerialize() {
		$out = array(
			'@context' => 'http://schema.org',
			'@type' => 'BusinessAudience'
		);
		
		$serialized = so_json_serialize( $this->number_of_employees );
		if ( ! empty( $serialized ) ) {
			$out['numberOfEmployees'] = $serialized;
		}
		
		$serialized = so_json_serialize( $this->yearly_revenue );
		if ( ! empty( $serialized ) ) {
			$out['yearlyRevenue'] = $serialized;
		}
		
		$serialized = so_json_serialize( $this->years_in_operation );
		if ( ! empty( $serialized ) ) {
			$out['yearsInOperation'] = $serialized;
		}
		
		return $out;
	}
}
